test/Grpc.AspNetCore.Server.Tests/PipeExtensionsTests.cs

```diff
@@ -0,0 +1,355 @@
+
+            // Assert 2
+            Assert.AreEqual(0, messageData2.Length);
+
+            // Act 3
+            var messageData3 = await pipeReader.ReadMessageStreamAsync();
+
+            // Assert 3
+            Assert.IsNull(messageData3);
+        }
+
+        [Test]
+        public void ReadMessageAsync_HeaderIncomplete_ThrowError()
+        {
+            // Arrange
+            var ms = new MemoryStream(new byte[]
+                {
+                    0x00, // compression = 0
+                    0x00,
+                    0x00
+                });
+
+            var pipeReader = new StreamPipeReader(ms);
+
+            // Act
+            var ex = Assert.ThrowsAsync<InvalidOperationException>(
+                () => pipeReader.ReadMessageAsync().AsTask());
+
+            // Assert
+            Assert.AreEqual("Incomplete message.", ex.Message);
+        }
+
+        [Test]
+        public void ReadMessageAsync_MessageDataIncomplete_ThrowError()
+        {
+            // Arrange
+            var ms = new MemoryStream(new byte[]
+                {
+                    0x00, // compression = 0
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x02, // length = 2
+                    0x10
+                });
+
+            var pipeReader = new StreamPipeReader(ms);
+
+            // Act
+            var ex = Assert.ThrowsAsync<InvalidOperationException>(
+                () => pipeReader.ReadMessageAsync().AsTask());
+
+            // Assert
+            Assert.AreEqual("Incomplete message.", ex.Message);
+        }
+
+        [Test]
+        public void ReadMessageStreamAsync_HeaderIncomplete_ThrowError()
+        {
+            // Arrange
+            var ms = new MemoryStream(new byte[]
+                {
+                    0x00, // compression = 0
+                    0x00,
+                    0x00
+                });
+
+            var pipeReader = new StreamPipeReader(ms);
+
+            // Act
+            var ex = Assert.ThrowsAsync<InvalidOperationException>(
+                () => pipeReader.ReadMessageAsync().AsTask());
+
+            // Assert
+            Assert.AreEqual("Incomplete message.", ex.Message);
+        }
+
+        [Test]
+        public void ReadMessageStreamAsync_MessageDataIncomplete_ThrowError()
+        {
+            // Arrange
+            var ms = new MemoryStream(new byte[]
+                {
+                    0x00, // compression = 0
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x02, // length = 2
+                    0x10
+                });
+
+            var pipeReader = new StreamPipeReader(ms);
+
+            // Act
+            var ex = Assert.ThrowsAsync<InvalidOperationException>(
+                () => pipeReader.ReadMessageStreamAsync().AsTask());
+
+            // Assert
+            Assert.AreEqual("Incomplete message.", ex.Message);
+        }
```

Comment:
Add a test for this condition? https://github.com/grpc/grpc-dotnet/pull/64/files#diff-feaffa8467755e7b6df8bf1465d5bed0R152